presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -391,7 +391,7 @@ protected Scope visitDelete(Delete node, Optional<Scope> scope)
                     session);
 
             Scope tableScope = analyzer.analyze(table, scope);
-            node.getWhere().ifPresent(where -> analyzer.analyzeWhere(node, tableScope, where));
+            node.getWhere().ifPresent(where -> analyzeWhere(node, tableScope, where));
```

Comment:
Can we make this change a separate commit? It seems to be an independent fix: enforce `SELECT` access checks for the `WHERE` clause.